### PR TITLE
Header Rewrite Policy support in IngressRoute

### PR DESCRIPTION
**Pull request by PinchasLev**
Add plumbing to IngressRoute to support header rewrite policies, matching the support that's already in Contour for HTTPProxy.

### CODE REVIEW

File: internal/e2e/adobe_test.go [hunk 1]
```diff
@@ -2237,3 +2240,158 @@ func TestAdobeRoute(t *testing.T) {
 		Nonce:       "1",
 	}, streamRDS(t, cc))
 }
+
+// == internal/envoy/route.go
+// add Route.RequestHeadersPolicy
+// add Route.ResponseHeadersPolicy
+
+func TestAdobeRouteHeaderRewritePolicy(t *testing.T) {
```
Comment (lrouquette) [inline internal/e2e/adobe_test.go:2248]: Awesome! nit: since this is for a CRD change, can you move this test to line ~780.

**Comment by PinchasLev** [comment]
@lrouquette After rebasing against v1.3-adobe to resolve the conflict to ADOBE_CHANGELOG.md, I see that some tests were removed from e2e/adobe_test.go.  Please confirm that was intended.

**Comment by lrouquette** [comment]
> @lrouquette After rebasing against v1.3-adobe to resolve the conflict to ADOBE_CHANGELOG.md, I see that some tests were removed from e2e/adobe_test.go. Please confirm that was intended.

Nope; your PR should not be removing tests (or change anything that's unrelated to header rewriting).

**Comment by PinchasLev** [comment]
> > @lrouquette After rebasing against v1.3-adobe to resolve the conflict to ADOBE_CHANGELOG.md, I see that some tests were removed from e2e/adobe_test.go. Please confirm that was intended.
> 
> Nope; your PR should not be removing tests (or change anything that's unrelated to header rewriting).

Fixed

**Pull request merged**